Tests/ApplicativeRouterTests/ApplicativeRouterTests.swift

```diff
@@ -110,24 +146,26 @@ class ApplicativeRouterTests: XCTestCase {
 
 enum Route {
   case home
-  case episodes
-  case episode(String)
+  case episodes(RefTag<Int?>)
```

Comment:
I wanted to experiment with how to derive more complicated parsers from simpler parsers. 

so, imagine you had a route that plucked out a bunch of data, and then in addition to that we want to pluck out a `ref` query param. we could include `ref: String?` in every route's associated value, _oooor_ we could create a wrapper type `RefTag<A> = (String?, a)` and allow _any_ route to be lifted to understand ref tags. That's what this route (and its tests) attempts to demonstrate.

This ref tag example is a total real life use case, tho I think instead of having `RefTag<A>` for every route we would flip it and have the router ultimately be a parser of `RefTag<Route>`. And really that might be generalized to a type `CommonData<A>` that wrapped `Route` and tried plucking out all values we expect to be common in all requests.

Fun!